### ISSUE TITLE
Documentation feature request: Explain Nesterov Accelerated Gradient implementation

### ISSUE THREAD

**Issue by armanschwarz** [tensorflow/tensorflow#19899]
I haven't filled out the provided form because this is a feature request for the documentation.

The documentation for [`tf.train.MomentumOptimizer`](https://www.tensorflow.org/api_docs/python/tf/train/MomentumOptimizer) offers a `use_nesterov` parameter on which the documentation says the following:

> `use_nesterov`: If True use Nesterov Momentum. See [Sutskever et al.,
> 2013](http://proceedings.mlr.press/v28/sutskever13.pdf). This
> implementation always computes gradients at the value of the
> variable(s) passed to the optimizer. Using Nesterov Momentum makes the
> variable(s) track the values called `theta_t + mu*v_t` in the paper.

The problem is that the linked paper outlines the normal NAG algorithm, which requires computation of a gradient at a different value to those provided (namely at the next step). This is not clarified in the documentation and led me to some confusion. I ended up [asking this question on StackOverflow](https://stackoverflow.com/questions/50774683/how-is-nesterovs-accelerated-gradient-descent-implemented-in-tensorflow) and cobbled together [an answer myself](https://stackoverflow.com/a/50774886/1613983), however I think the answer by `user1735003` is so complete that without too much wrangling it could greatly enhance the documentation:

[Answer by `user1735003`](https://stackoverflow.com/a/50778921/1613983)

Some clarification to the fact that tensorflow actually implements a modified version of the algorithm which is only correct under certain conditions would have been very helpful and would have saved me some time.

**Comment by tensorflowbutler** [comment]
Thank you for your post. We noticed you have not filled out the following field in the issue template. Could you update them if they are relevant in your case, or leave them as N/A? Thanks.
Have I written custom code
OS Platform and Distribution
TensorFlow installed from
TensorFlow version
Bazel version
CUDA/cuDNN version
GPU model and memory
Exact command to reproduce

**Comment by armanschwarz** [comment]
Have I written custom code: N/A
OS Platform and Distribution: N/A
TensorFlow installed from: N/A
TensorFlow version: 1.5
Bazel version: N/A
CUDA/cuDNN version: N/A
GPU model and memory: N/A
Exact command to reproduce: N/A
